src/fontlibc/fontlibc.h

```diff
@@ -0,0 +1,437 @@
+	intptr_t widths_table;
+	/* Offset to a table of offsets to glyph bitmaps.
+	 * These offsets are only 16-bits each to save some space. */
+	intptr_t bitmaps;
+	/* Specifies how much to move the cursor left after each glyph.
+	   Total movement is width - overhang.  Intended for italics. */
+	uint8_t italic_space_adjust;
+	/* These suggest adding blank space above or below each line of text.
+	   This can increase legibility. */
+	uint8_t space_above;
+	uint8_t space_below;
+	/* Specifies the boldness of the font. 
+	   0x40: light
+	   0x80: regular
+	   0x90: medium
+	   0xC0: bold*/
+	uint8_t weight;
+	/* Specifies the style of the font.  See enum font_styles */
+	uint8_t style;
+	/* For layout, allows aligning text of differing fonts vertically.
+	   These count pixels going down, i.e. 0 means the top of the glyph. */
+	uint8_t cap_height;
+	uint8_t x_height;
+	uint8_t baseline_height;
+} raw_font;
+
+typedef struct {
+	char header[8]; /* "FONTPACK" */
+	/* Frankly, if you have more than 127 fonts in a pack, you have a
+	   problem. */
+	uint8_t fontCount;
+	/* Offset from first byte of header */
+	intptr_t metadata;
+	intptr_t font_list;
+} font_pack;
+
+
+/**
+ * Sets the bounds of the window all text will appear in to be the full screen
+ */
+void fontlib_SetWindowFullScreen(void);
+
+/**
+ * Sets the bounds of the window all text will appear in
+ * @param x_min X coord base
+ * @param y_min Y coord base
+ * @param width Width
+ * @param height Height
+ */
+void fontlib_SetWindow(int x_min, uint8_t y_min, int width, uint8_t height);
+
+/**
+ * Returns the starting column of the current text window
+ * @return Window X
+ */
+int fontlib_GetWindowXMin(void);
+
+/**
+ * Returns the starting row of the current text window
+ * @return Window Y
+ */
+uint8_t fontlib_GetWindowYMin(void);
+
+/**
+ * Returns the width of the current text window
+ * @return Window width
+ */
+int fontlib_GetWindowWidth(void);
+
+/**
+ * Returns the height of the current text window
+ * @return Window height
+ */
+uint8_t fontlib_GetWindowHeight(void);
+
+/**
+ * Sets the cursor position
+ * @param x X
+ * @param y Y
+ */
+void fontlib_SetCursorPosition(int x, uint8_t y);
+
+/**
+ * Returns the cursor column.
+ * @return Current cursor X
+ */
+int fontlib_GetCursorX(void);
+
+/**
+ * Returns the cursor row.
+ * @return Current cursor Y
+ */
+uint8_t fontlib_GetCursorY(void);
+
+/**
+ * Adds the given (x,y) to the cursor position.
+ * Useful for tabbing, for example.
+ * @param x x-shift
+ * @param y y-shift
+ */
+void fontlib_ShiftCursorPosition(int x, uint8_t y);
+
+/**
+ * Sets the current font
+ * @param font_data Pointer to font data
+ * @param flags Information about how to process the font (unused)
+ * @return Returns false if the font seems invalid for any reason
+ * WARNING: If false is returned, no valid font is currently loaded and trying
+ * to print will print garbage!
+ */
+bool fontlib_SetFont(uint8_t* font_data, int flags);
```

Comment:
```suggestion
bool fontlib_SetFont(const fontlib_font_t *font, unsigned flags);
```

Reply:
The main idea was to add a flag for ignoring the space above/below metrics. But I'm calling it `unsigned int` :p

Reply:
Sure, `unsigned int` is fine too.